actions/access_keys/revoke.go

```diff
@@ -9,22 +9,42 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
+// revoke will revoke the intended model by id
+// Revoke access key godoc
+// @Summary		Revoke access key - Use (DELETE) /api/v1/users/current/keys/{id} instead.
+// @Description	This endpoint has been deprecated. Use (DELETE) /api/v1/users/current/keys/{id} instead.
+// @Tags		Access-key
+// @Produce		json
+// @Param		id query string true "id of the access key"
+// @Success		200	{object} models.AccessKey "Revoked AccessKey"
+// @Failure		400	"Bad request - Missing required field: id"
+// @Failure 	500	"Internal server error - Error while revoking access key"
+// @DeprecatedRouter  /v1/access-key [delete]
+// @Security	x-auth-xpub
+func (a *Action) oldRevoke(c *gin.Context) {
+	a.revokeHelper(c, true)
+}
+
 // revoke will revoke the intended model by id
 // Revoke access key godoc
 // @Summary		Revoke access key
 // @Description	Revoke access key
 // @Tags		Access-key
 // @Produce		json
-// @Param		id query string true "id of the access key"
+// @Param		id path string true "id of the access key"
 // @Success		200	{object} models.AccessKey "Revoked AccessKey"
 // @Failure		400	"Bad request - Missing required field: id"
 // @Failure 	500	"Internal server error - Error while revoking access key"
-// @Router		/v1/access-key [delete]
+// @Router		/api/v1/users/current/keys/{id} [delete]
 // @Security	x-auth-xpub
 func (a *Action) revoke(c *gin.Context) {
+	a.revokeHelper(c, false)
+}
+
+func (a *Action) revokeHelper(c *gin.Context, snakeCase bool) {
 	reqXPub := c.GetString(auth.ParamXPubKey)
 
-	id := c.Query("id")
+	id := c.Params.ByName("id")
```

Comment:
is this going to work for both: old and new endpoint?

Reply:
fixed